20_Simulation_Principles/60_Constraint/20_Lagrange_Constraint.md

```diff
@@ -1,70 +1,71 @@
+where $$\mathbf{W}=dt\left[\mathbf{H}_1\mathbf{A}_1^{-1}\mathbf{H}_1^T+\mathbf{H}_2\mathbf{A}_2^{-1}\mathbf{H}_2^T\right]$$ is the matrix of our linearized constraint system, this matrix $$\mathbf{W}$$ is homogeneous to a compliance. $$\dot{\delta}$$ is the constraint violation (here in velocity), that can be directly obtained from the expression of our constraint laws $$\Phi$$ and $$\Psi$$.
 
 Finally, the resolution of the constraint problem is done using the [Gauss-Seidel algorithm](https://en.wikipedia.org/wiki/Gauss%E2%80%93Seidel_method). After resolution of this new linear system, the motion can be corrected as follows:
 
-- <img class="latex" src="https://latex.codecogs.com/png.latex?x_1=x_1^{free}+dt\cdot%20\Delta%20v_1^{cor}" title="Correction1" />
-- <img class="latex" src="https://latex.codecogs.com/png.latex?x_1=x_2^{free}+dt\cdot%20\Delta%20v_2^{cor}" title="Correction2" /> 
-  with <img class="latex" src="https://latex.codecogs.com/png.latex?\Delta%20v_1^{cor}=dt\mathbf{A}_1^{-1}\mathbf{H}_1\lambda" title="Corrective displacement1" /> and <img class="latex" src="https://latex.codecogs.com/png.latex?\Delta%20v_2^{cor}=dt\mathbf{A}_2^{-1}\mathbf{H}_2\lambda" title="Corrective displacement2" />
+- $$x_1=x_1^{free}+dt\cdot%20\Delta%20v_1^{cor}$$
+- $$x_1=x_2^{free}+dt\cdot%20\Delta%20v_2^{cor}$$ 
+  with $$\Delta%20v_1^{cor}=dt\mathbf{A}_1^{-1}\mathbf{H}_1\lambda$$ and $$\Delta%20v_2^{cor}=dt\mathbf{A}_2^{-1}\mathbf{H}_2\lambda$$
 
 
 FreeMotionAnimationLoop
 -----------------
 
-To solve such complex constraint-based interactions, the simulation requires a specific [animation loop](https://www.sofa-framework.org/community/doc/simulation-principles/animation-loop/): the [_FreeMotionAnimationLoop_](https://www.sofa-framework.org/community/doc/components/animationloops/freemotionanimationloop/). This animation loop divides each simulation step into two successive resolution steps: the free motion and a corrective motion.
+To solve such complex constraint-based interactions, the simulation requires a specific [animation loop](../animation-loop/): the [_FreeMotionAnimationLoop_](../../components/animationloop/freemotionanimationloop/). This animation loop divides each simulation step into two successive resolution steps: the free motion and a corrective motion.
```

Comment:
Relativity is deep... Do we want to be canonical ?

Reply:
My bad, didn't catch the difference between animationloops and animation-loop at first glance. So you are not going back to come back in the same folder again as I though.